File: week01/kjg/네트워크.js



function solution(n,computers){
    let count = 0
    const visited = new Array(n).fill(false)

    function dfs(idx){

        visited[idx] = true // 방문처리

        for(let j = 0; j<n ; j++){
            if(computers[idx][j] === 1 && !visited[j]){
                dfs(j)
            }
        }



    }

    for(let i = 0; i<n; i++){
        if(!visited[i]){
            dfs(i)
            count++
        }
    }

    return count


}

File: week02/ohy/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;

    private void dfs(int node, int[][] computers) {
        visited[node] = true;

        for (int i = 0; i < computers[node].length; i++) {
            if (!visited[i] && computers[node][i] == 1) {
                dfs(i, computers);
            }
        }
    }
    
    private void bfs(int node, int[][] computers) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(node);
        
        while(!q.isEmpty()) {
            int cur = q.poll();
            visited[cur] = true;
            
            for (int i = 0; i < computers[cur].length; i++) {
                if (!visited[i] && computers[cur][i] == 1) {
                    q.offer(i);
                }
            }
        }
    }

    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];

        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                bfs(i, computers);
                answer++;
            }
        }

        return answer;
    }
}
